### ISSUE TITLE
gradient accumulation for sentence transformers

### ISSUE THREAD

**Issue by hanselowski** [UKPLab/sentence-transformers#1206]
Hi @nreimers,
I assume because the performance of bi-encoders strongly depends on the batch size, training with gradient accumulation would make sense.

![105222047-3b006d00-5b5a-11eb-922a-a180d6ea004a](https://user-images.githubusercontent.com/29311022/137286214-5c57240d-5196-49cf-abe8-cecce718a603.png)

Is there a way to use gradient accumulation in sentence transformers?
Thanks Andreas

**Comment by nreimers** [comment]
Correct, gradient accumulation does not make sense. The performance increase comes from the higher number of negatives.

Have a look at the MarginMSE Loss. It works well with small batches and achieves better performances than MultipleNegativesRankingLoss.

I also talked here about MarginMSE (starting at about minute 20):
https://www.youtube.com/watch?v=XHY-3FzaLGc
